### ISSUE TITLE
ckpt_path in trainer accepts URIs to automatically load checkpoint from remote paths

### ISSUE THREAD

**Issue by aretor** [Lightning-AI/pytorch-lightning#19849]
### Description & Motivation

If I set up a Trainer with a `WandbLogger` and set `log_model=True`, I get that my model is saved locally and in a W&B server. 

If I want to retrieve the model from the server I have to use the W&B `use_artifact` methods and the download methods to first retrieve the model, as described [here](https://lightning.ai/docs/pytorch/stable/extensions/generated/lightning.pytorch.loggers.WandbLogger.html)

What I would like to have instead is to not specify the W&B logic explicitly and download the model by just passing a URI with the remote string, (e.g. `wandb://user/project/model-run_id:version`) as the `ckpt_path` in the `Trainer`, along with the `WandbLogger`.

In short, I would like to do something like that:

```
trainer = Trainer(logger=my_wandb_logger, "wanb://user/project/model-run_id:version")
```

### Pitch

Specify a URI to a remote resource, automatically download it, and load it in the `Trainer`.

### Alternatives

Write explicitly the W&B logic to download the model and then pass the checkpoint path to the `Trainer`.
